src/software/backend/radio_backend.cpp

```diff
@@ -33,5 +33,14 @@ void RadioBackend::receiveWorld(World world)
     Subject<World>::sendValueToObservers(world);
 }
 
+void RadioBackend::receiveRobotStatus(RobotStatus robot_status)
+{
+    SensorMsg sensor_msg;
+    TbotsRobotMsg robot_msg        = convertRobotStatusToTbotsRobotMsg(&robot_status);
+    TbotsRobotMsg* added_robot_msg = sensor_msg.add_tbots_robot_msg();
+    added_robot_msg                = &robot_msg;
```

Comment:
I think what's happening here is that we're setting `added_robot_msg` equal to a pointer to the stack, which means it can be overwritten once this function goes out of scope. You need to make robot_msg a proper heap-allocated object

Reply:
Maybe its worth making `convertRobotStatusToTbotsRobotMsg` return a smart pointer to the allocated TbotsRobotMsg, then you can do the same `set_allocated_tbots_robot_msg` call that you did inside the function it self

Reply:
actually another thought, why not have `convertRobotStatustoSensorMsg` and then test that the repeated field is working as well?

Reply:
If I remember correctly, `set_allocated_tbots_robot_msg` doesn't exist as a method in `sensor_msg.pb.h`. That's why I had to do this method instead.

Reply:
Yeah returning a SensorMsg proto would be quite nice for testing

Reply:
So instead of `convertRobotStatustoTbotsRobotMsg`, you want `convertRobotStatustoSensorMsg` instead?

Reply:
@MathewMacDougall @garethellis0 Thoughts?

Reply:
If I'm understanding this right I think you could do
```
*added_robot_msg                = robot_msg;
```
to dereference the pointer and assign it's value.

IMO we should keep this function as `convertRobotStatusToTbotsRobotMsg` because that's all it does. One `RobotStatus` message is a report for a single robot, so I think it would be better to have a separate function for convcrting a list of `RobotStatus` messages into a `SensorMsg` with the repeated field if needed.